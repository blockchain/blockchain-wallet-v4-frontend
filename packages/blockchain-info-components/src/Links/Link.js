import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { Palette } from '../Colors'
import { keysIn } from 'ramda'

const BaseLink = styled.a`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  font-family: 'Montserrat', sans-serif;
  font-size: ${props => props.size};
  font-weight: ${props => props.weight};
  color: ${props => props.theme[props.color]};
  text-transform: ${props =>
    props.uppercase ? 'uppercase'
      : props.capitalize ? 'capitalize' : 'none'};
  text-decoration: none;
  cursor : pointer;

  &:hover {
    color: ${props => props.theme[props.color]};
  }

  &:focus { text-decoration: none; }
`

const Link = props => {
  const { children, ...rest } = props

  return (
    <BaseLink {...rest}>
      {children}
    </BaseLink>
  )
}

Link.propTypes = {
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900]),
  size: PropTypes.string,
  color: PropTypes.oneOf(keysIn(Palette())),
  uppercase: PropTypes.bool,
  capitalize: PropTypes.bool,
  bold: PropTypes.bool
}

Link.defaultProps = {
  weight: 400,
  size: '16px',
  color: 'brand-secondary',
  uppercase: false,
  capitalize: false,
  bold: false
}

export default Link
