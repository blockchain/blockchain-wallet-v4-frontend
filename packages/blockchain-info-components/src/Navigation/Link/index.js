import React from 'react'
import { injectIntl } from 'react-intl'

import { LinkEvent } from '../Events'

const Link = props => {
  let { children, href, locale, ...rest } = props
  let parsedURL = null
  let to = href
  let doReplace = false

  try {
    parsedURL = new URL(to)
  } catch (error) {
    parsedURL = null
  }
  if (parsedURL === null && !href.startsWith('#')) {
    doReplace = true
  }
  // insert locale in front of link only if its relative and locale is not "en"
  if (doReplace && props.intl && locale && locale !== 'en') {
    to = '/' + locale + (href.startsWith('/') ? href : '/' + href)
  }

  let newProps = {
    ...rest,
    href: to
  }
  if (props.event) {
    return <LinkEvent {...newProps}>{children}</LinkEvent>
  } else {
    return <a {...newProps}>{children}</a>
  }
}

export default injectIntl(Link)
