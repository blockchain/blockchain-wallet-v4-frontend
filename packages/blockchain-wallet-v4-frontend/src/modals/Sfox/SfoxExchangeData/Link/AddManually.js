import React, { Component } from 'react'
import styled from 'styled-components'
import { Field } from 'redux-form'
import { FormattedMessage } from 'react-intl'
import { TextBox, SelectBoxBankAccountType, Form } from 'components/Form'
import { required } from 'services/FormHelper'
import { Image, Text } from 'blockchain-info-components'
import { spacing } from 'services/StyleService'
import PropTypes from 'prop-types'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  margin-top: 25px;
  button {
    margin-top: 20px;
  }
`
const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  button {
    margin-top: 0px;
  }
`
const CheckImage = styled(Image)`
  width: 100%;
`

class AddManually extends Component {
  render () {
    const {
      handleFullName,
      handleAccountNumber,
      handleRoutingNumber,
      handleAccountType
    } = this.props
    return (
      <Form>
        <Container>
          <InputContainer>
            <Text size='14px' weight={500} style={spacing('mb-10')}>
              <FormattedMessage
                id='sfoxexchangedata.link.addmanually.accountholdername'
                defaultMessage='Full Name of Primary Account Holder'
              />
            </Text>
            <Field
              name='fullName'
              component={TextBox}
              validate={[required]}
              onChange={e => handleFullName(e)}
              placeholder='John Doe'
            />
          </InputContainer>
          <InputContainer>
            <Text size='14px' weight={500} style={spacing('mb-10')}>
              <FormattedMessage
                id='sfoxexchangedata.link.addmanually.bankaccountinformation'
                defaultMessage='Bank Account Information'
              />
            </Text>
            <CheckImage name='check-helper' />
            <Field
              name='routingNumber'
              component={TextBox}
              validate={[required]}
              onChange={e => handleRoutingNumber(e)}
              placeholder='Routing Number'
            />
            <Field
              name='accountNumber'
              component={TextBox}
              validate={[required]}
              onChange={e => handleAccountNumber(e)}
              placeholder='Account Number'
            />
          </InputContainer>
          <InputContainer>
            <Text size='14px' weight={500} style={spacing('mb-10')}>
              <FormattedMessage
                id='sfoxexchangedata.link.addmanually.accounttype'
                defaultMessage='Account Type'
              />
            </Text>
            <Field
              name='type'
              component={SelectBoxBankAccountType}
              validate={[required]}
              onChange={(e, val) => handleAccountType(e, val)}
            />
          </InputContainer>
        </Container>
      </Form>
    )
  }
}

AddManually.propTypes = {
  handleFullName: PropTypes.func.isRequired,
  handleAccountNumber: PropTypes.func.isRequired,
  handleRoutingNumber: PropTypes.func.isRequired,
  handleAccountType: PropTypes.func.isRequired
}

export default AddManually
