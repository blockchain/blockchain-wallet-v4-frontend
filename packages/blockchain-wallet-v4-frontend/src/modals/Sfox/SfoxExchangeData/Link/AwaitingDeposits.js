import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { Image, Text, Button } from 'blockchain-info-components'
import PropTypes from 'prop-types'
import media from 'services/ResponsiveService'
import { PartnerHeader } from 'components/IdentityVerification'

const Container = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  ${media.mobile`
    flex-direction: column;
    height: 100vh;
  `};
`
const LeftCol = styled.div`
  display: flex;
  flex-direction: column;
  width: 60%;
  > div {
    margin: 15px 0px;
  }
  ${media.mobile`
    width: 100%;
  `};
`
const RightCol = styled.div`
  width: 40%;
  display: flex;
  justify-content: center;
  ${media.mobile`
    width: 100%;
  `};
`
const ImageWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`
const DepositsImage = styled(Image)`
  width: 215px;
  ${media.mobile`
    width: 150px;
  `};
`

const AwaitingDeposits = ({ showModal }) => {
  return (
    <Container>
      <LeftCol>
        <PartnerHeader>
          <FormattedMessage
            id='modals.sfoxexchangedata.link.awaitingdeposits.title'
            defaultMessage='Verify Your Bank Account'
          />
        </PartnerHeader>
        <ImageWrapper>
          <DepositsImage name='micro-deposits-whole' />
        </ImageWrapper>
        <Text size='13px' weight={300}>
          <FormattedMessage
            id='modals.sfoxexchangedata.link.awaitingdeposits.body'
            defaultMessage='To verify your bank details, SFOX will send two micro-deposits to your bank account for a few cents each. Once received, select Enter Deposit Details to finish setting up your account.'
          />
        </Text>
        <Text size='13px' weight={300}>
          <FormattedMessage
            id='modals.sfoxexchangedata.link.awaitingdeposits.note'
            defaultMessage='Bear with us: receiving these deposits can take up to 5 business days.'
          />
        </Text>
      </LeftCol>
      <RightCol>
        <Button
          width='80%'
          nature='primary'
          onClick={() => showModal('SfoxEnterMicroDeposits')}
        >
          <FormattedMessage
            id='awaiting_deposits.enter'
            defaultMessage='Enter Deposit Details'
          />
        </Button>
      </RightCol>
    </Container>
  )
}

AwaitingDeposits.propTypes = {
  showModal: PropTypes.func.isRequired
}

export default AwaitingDeposits
