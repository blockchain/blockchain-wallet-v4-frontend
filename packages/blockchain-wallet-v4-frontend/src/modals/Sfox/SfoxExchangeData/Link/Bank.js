import React from 'react'
import styled from 'styled-components'
import { Text } from 'blockchain-info-components'
import PropTypes from 'prop-types'
import { prop } from 'ramda'

const BankInfo = styled.div`
  display: flex;
  flex-direction: row;
  padding: 25px 30px;
  border: 1px solid #ddd;
  input[type='radio'] {
    display: none;
  }
  input[type='radio']:checked + label:before {
    content: '';
    width: 16px;
    height: 16px;
    background-color: #004a7c;
  }
  margin: 0px;
`
const BankLabel = styled.label`
  display: flex;
  flex-direction: row;
  cursor: pointer;
  ::before {
    content: '';
    width: 16px;
    height: 16px;
    min-width: 16px;
    border-radius: 8px;
    background-color: white;
    border: 1px solid #004a7c;
    -webkit-transition: background-color 0.2s;
    -o-transition: background-color 0.2s;
    transition: background-color 0.2s;
  }
`
const BankMeta = styled.div`
  margin-left: 15px;
  display: flex;
  flex-direction: column;
  cursor: pointer;
`

const Bank = ({ bank, onInputClick }) => (
  <BankInfo key={prop('account_id', bank)}>
    <input
      type='radio'
      name='bank-filled'
      value={bank}
      id={prop('account_id', bank)}
      onClick={() => onInputClick(prop('account_id', bank))}
    />
    <BankLabel htmlFor={prop('account_id', bank)}>
      <BankMeta>
        <Text size='16px'>
          {prop('name', bank)} ending in {prop('mask', bank)}
        </Text>
      </BankMeta>
    </BankLabel>
  </BankInfo>
)

Bank.propTypes = {
  bank: PropTypes.shape({
    account_id: PropTypes.number,
    name: PropTypes.string,
    mask: PropTypes.number
  })
}

export default Bank
