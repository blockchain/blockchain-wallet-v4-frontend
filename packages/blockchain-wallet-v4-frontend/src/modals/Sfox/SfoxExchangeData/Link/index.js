import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Link from './template'
import { actions } from 'data'
import { merge, path, prop, head } from 'ramda'
import { Remote } from 'blockchain-wallet-v4/src'
import { getData } from './selectors'

class LinkContainer extends PureComponent {
  state = {
    enablePlaid: false,
    id: '',
    addBankManually: false,
    selectBank: false,
    microDeposits: false,
    microStep: 'welcome'
  }

  componentDidMount () {
    let receiveMessage = e => {
      const plaidWhitelist = ['enablePlaid', 'disablePlaid', 'getBankAccounts']
      if (!e.data.command) return
      if (e.data.from !== 'plaid') return
      if (e.data.to !== 'exchange') return
      if (e.origin !== this.props.plaidBaseUrl) return
      if (plaidWhitelist.indexOf(e.data.command) < 0) return

      if (e.data.command === 'enablePlaid') this.setState({ enablePlaid: true })
      if (e.data.command === 'disablePlaid') {
        this.setState({ enablePlaid: false })
      }
      if (e.data.command === 'getBankAccounts' && e.data.msg) {
        this.props.sfoxDataActions.getBankAccounts(e.data.msg)
        this.setState({
          enablePlaid: false,
          token: e.data.msg,
          selectBank: true
        })
      }
    }
    window.addEventListener('message', receiveMessage, false)
  }

  componentWillUnmount () {
    this.setState({
      addBankManually: false,
      selectBank: false,
      microDeposits: false,
      microStep: 'welcome'
    })
    this.props.sfoxDataActions.wipeBankAccounts()
  }

  resetAccountHolder = () => {
    this.props.formActions.reset('sfoxLink')
    this.props.sfoxFrontendActions.sfoxNotAsked()
  }

  onSetBankAccount = data => {
    const bankChoice = merge(data, { token: this.state.token })
    this.props.sfoxFrontendActions.setBankAccount(bankChoice)
  }

  submitMicroDeposits = () => {
    const amount1 = parseFloat(this.props.deposit1)
    const amount2 = parseFloat(this.props.deposit2)
    this.props.sfoxFrontendActions.submitMicroDeposits({ amount1, amount2 })
  }

  onSubmit = () => {
    if (
      this.state.addBankManually &&
      this.state.routingNumber &&
      this.state.accountNumber
    ) {
      this.props.sfoxFrontendActions.sfoxLoading()
      const { fullName, routingNumber, accountNumber, accountType } = this.state
      this.props.sfoxFrontendActions.setBankManually(
        routingNumber,
        accountNumber,
        fullName,
        accountType
      )
    } else {
      this.props.sfoxFrontendActions.sfoxLoading()
      const bankChoice = merge(
        {
          id: this.state.id,
          firstname: this.props.accountHolderFirst,
          lastname: this.props.accountHolderLast
        },
        { token: this.state.token }
      )
      this.props.sfoxFrontendActions.setBankAccount(bankChoice)
    }
  }

  render () {
    const {
      sfoxFrontendActions,
      modalActions,
      accounts,
      bankAccounts,
      busyStatus,
      plaidBaseUrl,
      plaidKey,
      plaidEnv
    } = this.props
    const { sfoxNotAsked } = sfoxFrontendActions
    const { showModal } = modalActions
    const plaidUrl = `${plaidBaseUrl}/wallet-helper/plaid/#/key/${plaidKey}/env/${plaidEnv}`

    let awaitingDeposits = false
    if (Remote.Success.is(accounts)) {
      awaitingDeposits =
        prop('status', head(accounts.getOrElse())) === 'pending'
    }

    const { sfoxBusy, err } = busyStatus.cata({
      Success: () => ({ sfoxBusy: false }),
      Loading: () => ({ sfoxBusy: true }),
      Failure: err => ({ sfoxBusy: false, err }),
      NotAsked: () => ({ sfoxBusy: false })
    })

    return (
      <Link
        onSubmit={this.onSubmit}
        plaidUrl={plaidUrl}
        enablePlaid={this.state.enablePlaid}
        bankAccounts={bankAccounts}
        accounts={accounts}
        onSetBankAccount={this.onSetBankAccount}
        toggleManual={() =>
          this.setState({ addBankManually: !this.state.addBankManually })
        }
        handleBankSelection={id => this.setState({ id })}
        onNameChange={name => this.setState({ holderName: name })}
        handleFullName={e => this.setState({ fullName: e.target.value })}
        handleRoutingNumber={e =>
          this.setState({ routingNumber: e.target.value })
        }
        handleAccountNumber={e =>
          this.setState({ accountNumber: e.target.value })
        }
        handleAccountType={(e, val) => this.setState({ accountType: val })}
        microStep={this.state.microStep}
        goToMicroDepositStep={step => this.setState({ microStep: step })}
        submitMicroDeposits={this.submitMicroDeposits}
        awaitingDeposits={awaitingDeposits}
        showModal={showModal}
        busy={sfoxBusy}
        setNotAsked={sfoxNotAsked}
        linkError={err && path(['message'], err)}
        resetAccountHolder={this.resetAccountHolder}
        {...this.state}
      />
    )
  }
}

const mapStateToProps = getData

const mapDispatchToProps = dispatch => ({
  formActions: bindActionCreators(actions.form, dispatch),
  sfoxDataActions: bindActionCreators(actions.core.data.sfox, dispatch),
  sfoxFrontendActions: bindActionCreators(actions.modules.sfox, dispatch),
  modalActions: bindActionCreators(actions.modals, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LinkContainer)
