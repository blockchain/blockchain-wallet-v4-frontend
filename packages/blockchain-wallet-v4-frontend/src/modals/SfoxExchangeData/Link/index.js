import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators, compose } from 'redux'
import ui from 'redux-ui'
import Link from './template'
import { actions, selectors } from 'data'
import { formValueSelector } from 'redux-form'

import { merge } from 'ramda'

class LinkContainer extends Component {
  constructor (props) {
    super(props)
    this.onSetBankAccount = this.onSetBankAccount.bind(this)
    this.onSubmit = this.onSubmit.bind(this)
    this.submitMicroDeposits = this.submitMicroDeposits.bind(this)

    this.state = { enablePlaid: false, id: '' }
  }

  componentDidMount () {
    let receiveMessage = (e) => {
      const plaidWhitelist = ['enablePlaid', 'disablePlaid', 'getBankAccounts']
      if (!e.data.command) return
      if (e.data.from !== 'plaid') return
      if (e.data.to !== 'exchange') return
      if (e.origin !== `http://localhost:8081`) return // TODO: get from wallet options
      if (plaidWhitelist.indexOf(e.data.command) < 0) return

      if (e.data.command === 'enablePlaid') this.setState({ enablePlaid: true })
      if (e.data.command === 'disablePlaid') this.setState({ enablePlaid: false })
      if (e.data.command === 'getBankAccounts' && e.data.msg) {
        this.props.sfoxDataActions.getBankAccounts(e.data.msg)
        this.setState({ enablePlaid: false, token: e.data.msg })
        this.props.updateUI({ selectBank: true })
      }
    }
    window.addEventListener('message', receiveMessage, false)

    if (this.props.accounts.length) {
      if (this.props.accounts[0].status === 'pending') this.props.updateUI({ microDeposits: true })
    }
  }

  onSetBankAccount (data) {
    const bankChoice = merge(data, {token: this.state.token})
    this.props.sfoxFrontendActions.setBankAccount(bankChoice)
  }

  submitMicroDeposits () {
    const amount1 = parseFloat(this.props.deposit1)
    const amount2 = parseFloat(this.props.deposit2)
    this.props.sfoxFrontendActions.submitMicroDeposits({ amount1, amount2 })
  }

  onSubmit (e) {
    e.preventDefault()
    if (this.props.ui.toggleManual && this.state.routingNumber && this.state.accountNumber) {
      this.props.updateUI({ busy: true })
      const { fullName, routingNumber, accountNumber, accountType } = this.state
      this.props.sfoxFrontendActions.setBankManually(routingNumber, accountNumber, fullName, accountType)
    } else {
      this.props.updateUI({ busy: true })
      const bankChoice = merge({ id: this.state.id, name: this.state.holderName }, {token: this.state.token})
      this.props.sfoxFrontendActions.setBankAccount(bankChoice)
    }
  }

  render () {
    const { plaidUrl, bankAccounts, accounts, ui } = this.props

    return <Link
      onSubmit={this.onSubmit}
      plaidUrl={plaidUrl}
      enablePlaid={this.state.enablePlaid}
      bankAccounts={bankAccounts}
      accounts={accounts}
      onSetBankAccount={this.onSetBankAccount}
      toggleManual={() => this.props.updateUI({ toggleManual: true })}
      ui={ui}
      handleBankSelection={(id) => this.setState({ id })}
      onNameChange={(name) => this.setState({ holderName: name })}
      handleFullName={(e) => this.setState({ fullName: e.target.value })}
      handleRoutingNumber={(e) => this.setState({ routingNumber: e.target.value })}
      handleAccountNumber={(e) => this.setState({ accountNumber: e.target.value })}
      handleAccountType={(e, val) => this.setState({ accountType: val })}
      microStep={ui.microStep}
      goToMicroDepositStep={(step) => this.props.updateUI({ microStep: step })}
      submitMicroDeposits={this.submitMicroDeposits}
    />
  }
}

const mapStateToProps = (state) => ({
  plaidUrl: 'http://localhost:8081/wallet-helper/plaid/#/key/0b041cd9e9fbf1e7d93a0d5a39f5b9/env/production', // TODO: get from wallet options
  bankAccounts: selectors.core.data.sfox.getBankAccounts(state),
  accounts: selectors.core.data.sfox.getAccounts(state).data,
  deposit1: formValueSelector('sfoxLink')(state, 'deposit1'),
  deposit2: formValueSelector('sfoxLink')(state, 'deposit2')
})

const mapDispatchToProps = (dispatch) => ({
  formActions: bindActionCreators(actions.form, dispatch),
  sfoxDataActions: bindActionCreators(actions.core.data.sfox, dispatch),
  sfoxFrontendActions: bindActionCreators(actions.modules.sfox, dispatch),
  modalActions: bindActionCreators(actions.modals, dispatch)
})

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  ui({ state: { toggleManual: false, selectBank: false, microDeposits: false, microStep: 'welcome', busy: false } })
)

export default enhance(LinkContainer)
