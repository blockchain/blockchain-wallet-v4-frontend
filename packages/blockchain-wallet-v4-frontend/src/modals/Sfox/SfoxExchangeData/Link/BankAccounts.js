import React, { Component } from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { reduxForm, Field } from 'redux-form'
import { TextBox, Form, FormGroup, FormItem, FormLabel } from 'components/Form'
import { required } from 'services/FormHelper'
import Bank from './Bank'
import { Text } from 'blockchain-info-components'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  margin-top: 15px;
  button {
    margin-top: 20px;
  }
`
const HolderText = styled(Text)`
  margin-top: 20px;
  margin-bottom: 5px;
`

class BankAccounts extends Component {
  constructor (props) {
    super(props)
    this.onInputClick = this.onInputClick.bind(this)
    this.onInputChange = this.onInputChange.bind(this)
  }

  onInputClick (id) {
    this.props.onBankSelection(id)
  }

  onInputChange (e) {
    this.props.handleNameChange(e.target.value)
  }

  render () {
    return (
      <Container>
        <Form>
          {this.props.data.map(bank => {
            return <Bank bank={bank} onInputClick={this.onInputClick} />
          })}
          <HolderText size='18px' weight={400}>
            <FormattedMessage
              id='sfoxexchangedata.link.accountholder'
              defaultMessage='Account Holder'
            />
          </HolderText>
          <FormGroup inline>
            <FormItem>
              <FormLabel for='accountHolderFirst'>
                <FormattedMessage
                  id='sfoxexchangedata.link.accountholderfirstname'
                  defaultMessage='First Name'
                />
              </FormLabel>
              <Field
                name='accountHolderFirst'
                component={TextBox}
                validate={required}
              />
            </FormItem>
            <FormItem>
              <FormLabel for='accountHolderLast'>
                <FormattedMessage
                  id='sfoxexchangedata.link.accountholderlastname'
                  defaultMessage='Last Name'
                />
              </FormLabel>
              <Field
                name='accountHolderLast'
                component={TextBox}
                validate={required}
              />
            </FormItem>
          </FormGroup>
        </Form>
      </Container>
    )
  }
}

export default reduxForm({ form: 'sfoxLink' })(BankAccounts)
