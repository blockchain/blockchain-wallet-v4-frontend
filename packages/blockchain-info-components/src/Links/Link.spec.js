import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Link from './Link'

describe('Link component', () => {
  it('default renders correctly', () => {
    const component = shallow(
      <Link>
        <span>tab</span>
      </Link>
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
