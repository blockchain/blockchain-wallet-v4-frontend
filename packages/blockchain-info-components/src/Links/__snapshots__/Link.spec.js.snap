// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component default renders correctly 1`] = `
<Link__BaseLink
  bold={false}
  capitalize={false}
  color="brand-secondary"
  size="16px"
  uppercase={false}
  weight={400}
>
  <span>
    tab
  </span>
</Link__BaseLink>
`;
